### PR TITLE
fix: [select-dialog] select dialog name filter format item show error

### PR DESCRIPTION
**Pull request by GongHeng2017**
1. remove useless code of name filter function
2. add logic code of after set namefilter, immediately refresh
3. when set SingleSelection, should not support multiple choices

Log: fix issue
Bug: https://pms.uniontech.com/bug-view-197323.html

### CODE REVIEW

File: src/plugins/filemanager/core/dfmplugin-workspace/models/fileitemdata.h [hunk 1]
```diff
@@ -27,11 +27,14 @@ class FileItemData
 
     QVariant data(int role) const;
 
+    void setAvailableState(bool b);
```
Comment (Clauszy) [inline src/plugins/filemanager/core/dfmplugin-workspace/models/fileitemdata.h:30]: setAvailable

File: src/plugins/filemanager/core/dfmplugin-workspace/models/fileitemdata.h [hunk 2]
```diff
@@ -27,11 +27,14 @@ class FileItemData
...
+
 private:
     FileItemData *parent { nullptr };
     QUrl url;
     FileInfoPointer info { nullptr };
     SortInfoPointer sortInfo { nullptr };
+    bool isAvailable { true };
```
Comment (Clauszy) [inline src/plugins/filemanager/core/dfmplugin-workspace/models/fileitemdata.h:37]: available

**Review by Clauszy — CHANGES_REQUESTED** [review]

**Review by Clauszy — DISMISSED** [review]

**Review by liyigang1 — APPROVED** [review]

**Review by Lighto-Ku — APPROVED** [review]

**Review by max-lvs — APPROVED** [review]

**Pull request merged**